core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala

```diff
@@ -680,9 +680,14 @@ class TopicCommandWithAdminClientTest extends KafkaServerTestHarness with Loggin
     assertTrue(simpleDescribeOutputRows(0).startsWith(s"Topic: $testTopicName"))
     assertEquals(2, simpleDescribeOutputRows.size)
 
-    val underReplicatedOutput = TestUtils.grabConsoleOutput(
-      topicService.describeTopic(new TopicCommandOptions(Array("--under-replicated-partitions"))))
-    assertEquals("--under-replicated-partitions shouldn't return anything", "", underReplicatedOutput)
+    TestUtils.waitUntilTrue(
+      () => {
+        val underReplicatedOutput = TestUtils.grabConsoleOutput(
+          topicService.describeTopic(new TopicCommandOptions(Array("--under-replicated-partitions"))))
```

Comment:
Why would we have URPs in the first place? Reassignment should not cause URPs now, right?

Reply:
Good point. We are actually seeing this in 2.4. I just took at look at that release and it looks like that release doesn't have the fix. Should I target this change to 2.4 only?

Reply:
Sure.

Reply:
What I just said is not true. Let me get back to you.

Reply:
You were correct. I think there was a small issue in how we determined the replication factor and if a partition was under reassignment.